torchgeo/datasets/ssl4eo.py

```diff
@@ -157,6 +221,62 @@ def __len__(self) -> int:
         """
         return len(self.scenes)
 
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        path = os.path.join(self.subdir, "*", "*", "all_bands.tif")
+        if glob.glob(path):
+            return
+
+        # Check if the tar.gz files have already been downloaded
+        exists = []
+        for suffix in self.checksums[self.split]:
+            path = self.subdir + f".tar.gz{suffix}"
+            exists.append(os.path.exists(path))
+
+        if all(exists):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+
+        for suffix, md5 in self.checksums[self.split].items():
+            download_url(
+                self.url.format(self.split, suffix),
+                self.root,
+                md5=md5 if self.checksum else None,
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+
+        # Concatenate all tarballs together
```

Comment:
I'm pretty confident that this won't work unless you have > 50 GB RAM on your system. Need to think about how to do this incrementally. I guess just read/write chunk-by-chunk. It's going to be very slow regardless.